File: SAMPLES/CPP/test_vardecl.cpp
// test_vardecl.cpp - Test TODO 10 : variables et affectations
// Declarations globales et locales, affectations simples et composees

int g = 42;
int h;
const int MAX = 100;

int main() {
    int x = 10;
    int y = 20;
    int z;
    z = x + y;
    x = x + 1;
    y = y - 1;
    int a = 5, b = 6, c = 7;
    a += b;
    c -= 2;
    x++;
    --y;
    return 0;
}

File: SAMPLES/VAXPASCAL/test_vardecl.pas
(* test_vardecl.pas - Test TODO 09 : declarations de variables et affectations *)
(* VAX Pascal variable declarations, attributes, assignments *)

program test_vardecl(output);

const
  MAX_SIZE = 100;
  PI_APPROX := 314;

var
  x : Integer;
  y, z : Integer;
  c : Char;
  b : Boolean;
  s : VARYING [80] OF CHAR;
  arr : ARRAY[1..10] OF Integer;
  p : ^Integer;

[GLOBAL] var
  shared_count : Integer;

[VOLATILE] var
  io_flag : Integer;

value
  x := 42;

begin
  x := 10;
  y := x;
  z := x + y;
  c := 'A';
  b := true;
end.
